std/json.d

```diff
@@ -359,6 +359,90 @@ struct JSONValue
         return type == JSONType.null_;
     }
 
+    /***
+     * Generic type value getter
+     * A convenience getter that returns this `JSONValue` as the specified D type.
+     * Note: only arithmetic, `bool`, `string`, `JSONValue` and `JSONValue[]` types are accepted
+     * Throws: `JSONException` if `T` cannot hold the contents of this `JSONValue`
+     */
+    @property inout(T) get(T)() inout const pure @safe
+    {
+        static if (is(T : string))
+        {
+            return str;
```

Comment:
If we want to implicitly support user defined types:

```suggestion
            return cast(T) str;
```

Reply:
As far as I understand everyone in this trhead we don't want to make tricky conversions. CC @CyberShadow for clearance.

Reply:
Fine by me, just a suggestion based on @PetarKirov's comment

> Off topic for this PR: I wonder what would be the best way to support user-defined types, such as bigint, decimal or checkedint, without whitelisting them one by one here.

Reply:
Gotcha

Reply:
Reading a JSON string, parsing it into a fixed-size number (`double` or `long`), then converting it to a `BigInt` is completely meaningless, and providing such a conversion path would be disingenious. We would have to store and convert the string to `BigInt` directly as per the discussion above. @PetarKirov